lua/sp.c

```diff
@@ -6555,11 +6564,22 @@ static int setup_sp_for_trigger(trigger_reg_t *reg, char **err,
 
     char *spname = reg->spname;
     Q4SP(qname, spname);
+
+    rdlock_schema_lk();
     struct dbtable *db = getqueuebyname(qname);
     if (db == NULL) {
         *err = strdup("getqueuebyname failed");
+        unlock_schema_lk();
         return -1;
     }
+    rc = bdb_lock_table_read_fromlid(db->handle,
+              bdb_get_lid_from_cursortran(clnt->dbtran.cursor_tran));
+    if (rc != 0) {
+        unlock_schema_lk();
+        return luaL_error(L, "cannot read-lock queue for sp:%s", spname);
+    }
+    unlock_schema_lk();
+
```

Comment:
Looks like repetition from `db_consumer`. Perhaps refactor to have one function to initialize?

Reply:
Good idea.  There is still much polishing to do on this pull request.